erpnext/setup/doctype/company/company.py

```diff
@@ -396,6 +401,13 @@ def on_trash(self):
 		if not frappe.db.get_value('GL Entry', {'company': self.name}):
 			frappe.db.sql("delete from `tabProcess Deferred Accounting` where company=%s", self.name)
 
+	def check_parent_changed(self):
+		frappe.flags.parent_company_changed = False
+
+		if not self.is_new() and \
+			self.parent_company != frappe.get_cached_value("Company",  self.name,  "parent_company"):
```

Comment:
```suggestion
			self.parent_company != frappe.db.get_value("Company",  self.name,  "parent_company"):
```

`frappe.get_cached_value` calls `get_doc` internally if cached doc is not found.